### PR TITLE
python - pr

### PR DESCRIPTION
**Pull request by scbedd**
This PR is the actual implementation of a expanding/contracting public pull request build.

`python - pr` will replace all existing `python - <service> - ci` builds in `public`.

To manually run the the `python - pr` build instead of for a PR, one can manually queue it and set the `service` to the package that should be scanned.

TODO:

- [x] Initial yml updates
- [x] Verify pull request conditions vs "auto" service directory argument
  - ~~Do we need to be worried about pull request logic showing up if we /azp run a job? Should I pull back to only honor the `auto` instead of `Build.Reason`?~~
  - No. @weshaggard would prefer that the same expandable logic (including beyond their package directory) is possible. This is now in place.
  - We also need to change this such that ServiceDirectory targeted also build dependent packages.
- [x] Handle `verify readme` error
- [x] Ensure that `verify changelog` static analysis actually runs (notice it's all skipped currently)
- [x] Handle issues with [detect api changes](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3995347&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=bd13cdb4-3be4-5630-c04e-f305277c068c)
- [x] Test an internal core invocation from a package, ensure that the expansion still works
- [x] Test a nightly livetest build to verify that nothing from previous methodology is broken -- [check](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4051829&view=results)
- [x] Test template release build -- [check](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4051831&view=results)
- [x] Test nightly scheduled build to verify package property updates

### CODE REVIEW

File: eng/pipelines/templates/steps/analyze.yml [hunk 1]
```diff
@@ -27,14 +27,20 @@ steps:
...
-          PackageName: ${{artifact.name}}
-          ServiceName: ${{parameters.ServiceDirectory}}
-          ForRelease: false
-          Condition: succeededOrFailed()
+  - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+    - template: /eng/common/pipelines/templates/steps/verify-changelogs.yml
+      parameters:
+        PackagePropertiesFolder: $(Build.ArtifactStagingDirectory)/PackageInfo
+        Condition: succeededOrFailed()
+  - ${{ else }}:
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:35]: Any reason to not use the new verify-changelogs for both cases?

File: eng/scripts/get_package_properties.py [hunk 2]
```diff
@@ -1,11 +1,74 @@
...
-import sys
-import glob
 import os
 import re
 
+from typing import Dict, List
+
 from ci_tools.parsing import ParsedSetup
 
+additional_pr_triggers: Dict[str, List[str]] = {
```
Comment (scbedd) [inline eng/scripts/get_package_properties.py:9]: This is the source of truth for dependent packages that the python team will maintain.

File: eng/pipelines/templates/steps/analyze.yml (hunk 1 above)
Comment (scbedd) [reply eng/pipelines/templates/steps/analyze.yml:35]: Nope!

**Comment by weshaggard** [comment]
@scbedd when you think you are ready can you call a meeting with @benbp and I so we can do a live review of all the changes. I want to make sure we both fully understand these.

**Comment by scbedd** [comment]
Closing to prevent pounding on the agent pool while I clean up python - pr from the latest updates.

**Pull request closed without merging**